### PR TITLE
fix: Audience should include client ID

### PR DESCRIPTION
**Pull request by mitar**
## Proposed changes

[The spec](https://openid.net/specs/openid-connect-core-1_0.html#UserInfo) says:

> The aud value SHOULD be or include the RP's Client ID value. 

So this makes sure client ID is always there. It is true though that that section talks only about the situation when response is signed.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

File: oauth2/handler.go
Comment (mitar) [inline oauth2/handler.go:311]: This was a bug, BTW.

**Comment by aeneasr** [comment]
Could you please add a test for this? Thank you!

File: oauth2/handler.go
Comment (aeneasr) [inline oauth2/handler.go:321]: It would probably make sense to type assert once and set the value on interim later. You might use ory/x/stringslice for convenience!

**Review by aeneasr — COMMENTED** [review]
LGTM! Could you please add a test for this? :)

**Comment by aeneasr** [comment]
Are you still up for the changes? :) If you need any help, let us know!

**Comment by aeneasr** [comment]
Converting to draft due to lack of response

File: oauth2/handler.go
Comment (mitar) [reply oauth2/handler.go:321]: Updated. I didn't find an opportunity to use ory/x/stringslice though.

**Comment by mitar** [comment]
I could not find any existing tests for this handler, so I do not know how to write a test for this. If somebody can help with this. Otherwise I think the PR is ready.

**Comment by aeneasr** [comment]
Basically you could copy&paste [this test case](https://github.com/ory/hydra/blob/master/oauth2/handler_test.go#L209-L235) two or three times and cover the different audience format scenarios here.

As a maintainer I have been bitten so often by contributions without proper tests - from bugs to security issues to regressions. That's why we're also going to change our contribution guidelines which demand tests and might even introduce some coverage tests that fail to pass if test coverage is reduced with contributions.

So unfortunately I can't merge this until proper tests have been added.

**Comment by mitar** [comment]
I completely understand about tests. Thanks for pointing them out.

**Comment by mitar** [comment]
I added tests. But I could not run them locally (see https://github.com/ory/hydra/issues/2532).

**Review by aeneasr — APPROVED** [review]
Thank you @mitar :)

**Pull request merged**
